ros2/src/airsim_ros_pkgs/src/airsim_ros_wrapper.cpp

```diff
@@ -1328,6 +1316,42 @@ void AirsimROSWrapper::lidar_timer_cb()
     }
 }
 
+std::shared_ptr<airsim_interfaces::msg::GimbalAngleEulerCmd> AirsimROSWrapper::get_gimbal_msg_from_camera(
+    const std::string& vehicle_name, const std::string& camera_name)
+{
+    auto vehicle_attitude = get_camera_attitude_euler_msg(vehicle_name, "front_center_fixed");
+    auto gimbal_attitude = get_camera_attitude_euler_msg(vehicle_name, camera_name);
+
+    auto message = std::make_shared<airsim_interfaces::msg::GimbalAngleEulerCmd>();
+    message->header.stamp = nh_->get_clock()->now();
+    message->roll = gimbal_attitude->roll;
+    message->pitch = gimbal_attitude->pitch;
+
+    // Calculate "vehicle relative" yaw of the gimbaled camera.
+    message->yaw = math_common::constrain_euler_angle_to_360(gimbal_attitude->yaw - vehicle_attitude->yaw);
+
+    return message;
+}
+
+std::shared_ptr<airsim_interfaces::msg::GimbalAngleEulerCmd> AirsimROSWrapper::get_camera_attitude_euler_msg(
+    const std::string& vehicle_name, const std::string& camera_name)
+{
+    auto message = std::make_shared<airsim_interfaces::msg::GimbalAngleEulerCmd>();
+    message->header.stamp = nh_->get_clock()->now();
+
+    // Get the RPY orientation of a designated fixed camera (i.e. the vehicle orientation).
+    msr::airlib::CameraInfo vehicle_camera_info = airsim_client_->simGetCameraInfo(camera_name, vehicle_name);
+    tf2::Matrix3x3(get_tf2_quat(vehicle_camera_info.pose.orientation)).getRPY(
+        message->roll, message->pitch, message->yaw);

```

Comment:
is the camera_info.pose.orientation global?

Reply:
Yes.